### ISSUE TITLE
Invalid param name

### ISSUE THREAD

**Issue by kobezzza** [google/closure-compiler#773]
```js
/**
 * @param {!Object} params - parameters
 * @param {number} params.foo - some value
 */
function compile (params) {}
```

```
JSC_INVALID_PARAM: Parse error. invalid param name "params.foo" at line 3 character 19
* @param {number} params.foo - some value
```

Earlier this syntax worked. This syntax is more convenient than

```js
/**
 * @param {{foo: number}} params - parameters
 */
function compile (params) {}
```

Because it allows to add a description of the parameters.

**Comment by MatrixFrog** [comment]
I don't think we ever supported it. The change is that we used to silently ignore it (meaning the user may believe we're typechecking that param but we're actually not) but now we warn.

I think `--jscomp_off=nonStandardJsDocs` (or `--jscomp_warning=nonStandardJsDocs`) will let the compile succeed.

However, we should consider supporting this since `function f({foo: foo}) { use(foo); }` is valid in ES6. So I'll mark this as an enhancement request.

**Comment by concavelenz** [comment]
I don't think the ES6 syntax has much to do with this.  The parameter would be of type "{foo: SomeType}" even if the function declaration is using destructuring.

**Comment by MatrixFrog** [comment]
True, I just meant this style might become more common as ES6 rolls out. But it's a nice style even in ES5 so I'm definitely not against supporting it.

**Comment by samccone** [comment]
this would be really nice to have, I find it to be slightly more readable than inline also +1 for descriptions per arg

**Comment by christiantjl** [comment]
Can confirm this is a bug - `-O WHITESPACE` should not be transpiling anything, just removing comments and whitespace. There should be no `$jscomp` injection at all.
